enhancements/20190903-SRIOV-GA.md

```diff
@@ -0,0 +1,178 @@
+title: SR-IOV GA
+authors:
+  - "@zshi-redhat"
+  - "@pliurh"
+  - "@s1061123"
+reviewers:
+  - "@fepan"
+  - "@dougbtv"
+  - "@vpickard"
+  - "@dcbw"
+  - "@squeed"
+approvers:
+  - "@derekwaynecarr"
+creation-date: 2019-09-03
+last-updated: 2019-09-03
+status: implementable
+---
+
+# Graduate SR-IOV to GA
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+A proposal to graduate SR-IOV support from Tech Preview to GA.
+
+SR-IOV networks are enabled by Multus on secondary-onward pod interfaces.
+It is supported as Tech Preview on OpenShift 4.1 and 4.2 releases.
+This proposal is to make SR-IOV Generally Available in 4.3.
+
+## Motivation
+
+* SR-IOV is currently Technical Preview, but customers need to build supported
+  solutions on the technology.
+
+### Goals
+
+- Enhance Multus to support runtimeConfig for IP assignment
+- Enhance Multus to handle default route overwrite
+- Add admission controller for SR-IOV Network Operator
+- Compare SR-IOV network performance w/wo NUMA affinity
+- Support SR-IOV QoS flags
+- Add SR-IOV Metrics and Alerting
+- Support both RHEL and RHCOS worker nodes
+
+### Non-Goals
+
+- GA userspace (dpdk) workload support
+
+## Proposal
+
+### Implementation Details
+
+#### Multus
+
+Static IPAM plugin is used to assign IPv4 and IPv6 addresses statically to
```

Comment:
This is going to be hard, because Multus has to ask for runtimeConfiguration via its CNI configuration file, which is static. We'll probably need to think of a way for NetworkAttachmentDefinitions to ask for CNI capabilites.

cc @dcbw

Reply:
So, the proposal is *just* about passing the IP runtimeConfig? That makes more sense.

Reply:
Right.